klient/k8s/yaml/yaml.go

```diff
@@ -0,0 +1,130 @@
+	if err != nil {
+		return nil, err
+	}
+	return append(yamls, append(ymls, jsons...)...), nil
+}
+
+// LoadUnstructuredYAMLDirectory loads the set of single-document YAML and JSON files (*.yaml and *.yml and *.json) at the given directory
+func LoadUnstructuredYAMLDirectory(directory string) ([]*unstructured.Unstructured, error) {
+	files, err := listResourceFiles(directory)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not list directory %q", directory)
+	}
+
+	objects := []*unstructured.Unstructured{}
+	for _, file := range files {
+		obj, err := LoadUnstructuredYAML(file)
+		if err != nil {
+			return objects, err
+		}
+		objects = append(objects, obj)
+	}
+	if len(objects) == 0 {
+		return nil, fmt.Errorf("no files found in dir: %s", directory)
+	}
+	return objects, nil
+}
+
+// CreateResourceFromYAML loads a YAML or JSON file and creates it using the given klient.Client
+func CreateResourceFromYAML(klient klient.Client, filename, namespace string, opts ...resources.CreateOption) error {
+	obj, err := LoadUnstructuredYAML(filename)
+	if err != nil {
+		return err
+	}
+	obj.SetNamespace(namespace)
+	err = klient.Resources(namespace).Create(context.TODO(), obj, opts...)
+	return errors.Wrapf(err, "error creating unstructured object: %q; kind: %q; namespace: %q; name: %q", filename, obj.GetKind(), obj.GetNamespace(), obj.GetName())
+}
+
+// CreateResourcesFromYAMLDirectory loads YAML or JSON files from a directory and creates them using the given klient.Client
+func CreateResourcesFromYAMLDirectory(klient klient.Client, directory, namespace string, opts ...resources.CreateOption) error {
+	files, err := listResourceFiles(directory)
+	if err != nil {
+		return errors.Wrapf(err, "could not list directory %q", directory)
+	}
+	for _, f := range files {
+		if err := CreateResourceFromYAML(klient, f, namespace, opts...); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteResourceFromYAML loads a YAML or JSON file and deletes it using the given klient.Client
+func DeleteResourceFromYAML(klient klient.Client, filename, namespace string, opts ...resources.DeleteOption) error {
+	obj, err := LoadUnstructuredYAML(filename)
+	if err != nil {
+		return err
+	}
+	obj.SetNamespace(namespace)
+	err = klient.Resources(namespace).Delete(context.TODO(), obj, opts...)
+	return errors.Wrapf(err, "error deleting unstructured object: %q; kind: %q; namespace: %q; name: %q", filename, obj.GetKind(), obj.GetNamespace(), obj.GetName())
+}
+
+// DeleteResourcesFromYAMLDirectory loads YAML or JSON files from a directory and deletes them using the given klient.Client
+func DeleteResourcesFromYAMLDirectory(klient klient.Client, directory, namespace string, opts ...resources.DeleteOption) error {
+	files, err := listResourceFiles(directory)
+	if err != nil {
+		return errors.Wrapf(err, "could not list directory %q", directory)
+	}
+	for _, f := range files {
+		if err := DeleteResourceFromYAML(klient, f, namespace, opts...); err != nil {
```

Comment:
When we support this kind of load from dir, do we need to worry about ordering ? something similar to how helm follows in ordering of application of a resource and cleanup of a resource ? Resource with Finalizers can deadlock this workflow if not followed in a certain order right ? (Or am I over thinking this whole thing ?)

Reply:
This is probably correct regarding finalizers -- for my purposes, I usually name the `testdata` yamls with an obvious order in mind; `testdata/01-....yaml`, `testdata/02-...yaml` and rely on the file-system listing-order.

I should probably add a sort to `listResourceFiles`; as right now they sorted by filename, but by file extension and filename.

If I need to interact with a set of resources before/after another, I create another directory

Reply:
@crandles  What you explained as your test structure also makes a lot of sense. Considering this is a set of test assessments being done, one would probably have a sequence of workflow that is being evaluated. And they would follow a series of steps in a definitive way. 

MIght be a good thing to call this out in the Read me + the Example's readme both about having to take additional precaution about deleting resource using collection of files in the dir.